File: src/test/java/com/yourcompany/tests/SignUpTest.java
package com.yourcompany.tests;

import com.microsoft.playwright.*;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;
import java.nio.file.Paths;
import java.security.SecureRandom;

public class SignUpTest {

    // Hàm tạo email ngẫu nhiên
    private static String getRandomEmail(String prefix) {
        SecureRandom random = new SecureRandom();
        return prefix + random.nextInt(1000000) + "@example.com";
    }

    // Hàm tạo mật khẩu mạnh
    private static String generateStrongPassword() {
        String chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:',.<>?";
        SecureRandom random = new SecureRandom();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 12; i++) { // tạo mật khẩu dài 12 ký tự
            int randomIndex = random.nextInt(chars.length());
            sb.append(chars.charAt(randomIndex));
        }
        return sb.toString();
    }

    @Test
    public void testSignUp() {
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setHeadless(false));
            Page page = browser.newPage();
            page.navigate("https://magento.softwaretestingboard.com/");
            page.click("text='Create an Account'");
            String randomEmail = getRandomEmail("user");
            String password = generateStrongPassword();

            page.fill("input[name='firstname']", "Tran");
            page.fill("input[name='lastname']", "Bao");
            page.fill("input[name='email']", randomEmail);
            page.fill("input[name='password']", password);
            page.fill("input[name='password_confirmation']", password);

            page.click("button[type='submit']:has-text('Create an Account')");
            page.waitForSelector("text='Thank you for registering with Main Website Store.'");
            System.out.println("Account Created for: " + randomEmail);

            page.screenshot(new Page.ScreenshotOptions().setPath(Paths.get("screenshotjavatest/SignUpTest.png")));
            browser.close();
        } catch (Exception e) {
            e.printStackTrace();
            Assertions.fail("Test failed with an exception: " + e.getMessage());
        }
    }
}

File: tests/main/signUpTest.js
const { chromium } = require('playwright');
const crypto = require('crypto');

function getRandomString(prefix) {
    return `${prefix}${crypto.randomBytes(3).toString('hex')}`;
}

function generateStrongPassword() {
    const lowerCase = "abcdefghijklmnopqrstuvwxyz";
    const upperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    const digits = "0123456789";
    const specialChars = "!@#$%^&*()_+-=[]{}|;:',.<>?";
    return crypto.randomBytes(3).toString('hex') +
        lowerCase[Math.floor(Math.random() * lowerCase.length)] +
        upperCase[Math.floor(Math.random() * upperCase.length)] +
        digits[Math.floor(Math.random() * digits.length)] +
        specialChars[Math.floor(Math.random() * specialChars.length)];
}

(async () => {
    const browser = await chromium.launch({ headless: false });
    const page = await browser.newPage();
    await page.goto('https://magento.softwaretestingboard.com/');
    await page.click('text="Create an Account"');

    const username = getRandomString('user');
    const password = generateStrongPassword();

    await page.fill('input[name="firstname"]', 'Tran');
    await page.fill('input[name="lastname"]', 'Bao');
    await page.fill('input[name="email"]', `${username}@example.com`);
    await page.fill('input[name="password"]', password);
    await page.fill('input[name="password_confirmation"]', password);
    await page.click('button.submit.primary[type="submit"]:has-text("Create an Account")');
    await page.screenshot({ path: 'C:\\Users\\Admin\\Desktop\\MagentoTest\\tests\\reports\\screenshots\\signUpTest.png' });
    await browser.close();
})();
